crates/fluvio/src/producer/partition_producer.rs

```diff
@@ -25,6 +38,7 @@ pub(crate) struct PartitionProducer {
     batches_lock: Arc<Mutex<VecDeque<ProducerBatch>>>,
     batch_events: Arc<BatchEvents>,
     last_error: Arc<RwLock<Option<ProducerError>>>,
+    client_stats: Option<Arc<RwLock<ClientStats>>>,
```

Comment:
The internal stat counter should use Atomic to be highly performant, Lock will be too slow.